swarm/testutil/http.go

```diff
@@ -49,24 +54,86 @@ func NewTestSwarmServer(t *testing.T) *TestSwarmServer {
 		ChunkStore: localStore,
 	}
 	dpa.Start()
-	a := api.NewApi(dpa, nil)
+
+	// mutable resources test setup
+	resourceDir, err := ioutil.TempDir("", "swarm-resource-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ipcPath := filepath.Join(resourceDir, "test.ipc")
+	ipcl, err := rpc.CreateIPCListener(ipcPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rpcServer := rpc.NewServer()
+	rpcServer.RegisterName("eth", &FakeRPC{})
+	go func() {
+		rpcServer.ServeListener(ipcl)
+	}()
+	rpcClean := func() {
+		rpcServer.Stop()
+	}
+
+	// connect to fake rpc
+	rpcClient, err := rpc.Dial(ipcPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ethClient := ethclient.NewClient(rpcClient)
```

Comment:
Is it really necessary to start an RPC server for this? Should the resource handler not just accept an interface containing the methods it needs?

Reply:
Very good point, sir.